apps/state/lib/state/route_pattern.ex

```diff
@@ -78,4 +88,8 @@ defmodule State.RoutePattern do
     |> Enum.map(& &1.route_pattern_id)
     |> Enum.uniq()
   end
+
+  defp canonical_comparison(_, nil), do: true
+  defp canonical_comparison(1, bool), do: bool
+  defp canonical_comparison(_, bool), do: !bool
```

Comment:
question: do we need to accept `null` as a filter value?

Reply:
so the possibilities this covers is if the client passes either nil or any non true/false value. The `filter_by` isn't hit if there's no `is_canonical` key, but nil, empty string, etc will all end up here as nil rather than just skipping. Following all the discussion that took place regarding this is kind of scattered, the `nil|true|false` definition was mentioned in the ticket's comments and some ad-hoc discussions